syne_tune/tuner_callback.py

```diff
@@ -145,3 +146,90 @@ def on_tuning_end(self):
         # store the results in case some results were not committed yet (since they are saved every
         # `results_update_interval` seconds)
         self.store_results()
+
+
+class TensorboardCallback(TunerCallback):
+    def __init__(
+            self,
+
+            metric_to_optimize: 'str',
+            mode: 'str',
+            output_metrics: List[str] = [],
```

Comment:
Should be `Optional[List[str]] = None` to avoid side-effects. 

Also, I think the default behavior should rather be that if this is None then all the report metrics are tracked (the user could still decide to track none by passing `[]`).

Also, passing the mode is a bit annoying especially if its mandatory. Maybe a good first version would be to log all metrics without cummin/cummax for now? (it is also closer to the typical usage of tensor-board).

Reply:
Good point. I changed this such that by default all metric are reported, but users could define which metric to ignore. I still think it would be quite handy to plot the cumulative optimum and the best hyperparameter found so far. I suggest to add an optional parameter, that, if set, will report the cumulative optimum to tensoboard.